api/users/tests/test_models.py

```diff
@@ -318,3 +323,55 @@ def test_user_create_does_not_call_pipedrive_tracking_if_ignored_domain(
 
 def test_user_email_domain_property():
     assert FFAdminUser(email="test@example.com").email_domain == "example.com"
+
+
+@pytest.mark.django_db
+def test_change_email_address_api(mocker):
```

Comment:
It has been designed to work exactly in that way, am I missing something?

Reply:
Yes, the issue is that, if you run this test like this, it will fail: 

```bash
TASK_RUN_METHOD=TASK_PROCESSOR pytest -k test_change_email_address_api 
```

The best option here would be to mock the `send_email_changed_notification_email` task function, not the `send_mail` function. So we can change the mock to: 

```python
mocked_task = mocker.patch("users.models.send_email_changed_notification_email")

...

mocked_task.delay.assert_called_once_with(...)
```

Then we would need to add a unit test for the task function. 

The general rule here is that we should never end to end test anything that expects a function decorated with `@register_task_handler` to be called as it could mean we get different behaviour in different test environments. We want to try to make our test suite replicable in all environments.